src/redux/services/resourceRefs.ts

```diff
@@ -510,3 +510,30 @@ export function processRefs(
 
   cleanResourceRefs(resources);
 }
+
+/**
+ * Return a list of resource ids to reprocess when the specified resource has been updated
+ */
+
+export function findResourcesToReprocess(resource: K8sResource, resourceMap: ResourceMapType) {
+  // the resource itself
+  let resources = [resource.id];
+
+  // all existing resources with incoming refs to the resource (since they might have broken)
+  resource.refs
+    ?.filter(ref => isIncomingRef(ref.type))
+    .forEach(ref => {
+      if (ref.target?.type === 'resource' && ref.target.resourceId) {
+        resources.push(ref.target.resourceId);
+      }
+    });
+
+  // all existing resources with broken refs (since they might now work) -> this should be improved to
```

Comment:
good point, should we refactor `reprocessResources` from `services/resource` to do a filtering like this instead of checking of for all dependent resource kinds and then reprocessing all resources of that kind?

Reply:
i was thinking we would add an optional list of targetResourceKinds to the unsatisfied ref (since we know them when we try to discover the ref) - which we could use here for further filtering!?

Reply:
actually - looking at the code it seems we already add this to the ref - will see how that works!

Reply:
no - this requires more work... will merge this and tackle that separately...